benchmarker/index.js

```diff
@@ -0,0 +1,317 @@
+  log.report('!!!');
+  log.report('!!!');
+  log.report('--------------------------');
+}
+
+function apiPostFile(path, filePath) {
+  const mimeType = mimetypeFor(filePath);
+  const blob = fileFromSync(filePath, mimeType);
+  return apiPost(path, blob, { 'Content-Type':mimeType });
+}
+
+function apiPostJson(path, body, headers) {
+  return apiPost(path, JSON.stringify(body), { 'Content-Type':'application/json', ...headers });
+}
+
+function apiGetAndDump(prefix, n, path, headers) {
+  return fetchToFile(prefix, n, 'GET', path, undefined, headers);
+}
+
+function apiPostAndDump(prefix, n, path, body, headers) {
+  return fetchToFile(prefix, n, 'POST', path, body, headers);
+}
+
+async function fetchToFile(filenamePrefix, n, method, path, body, headers) {
+  const res = await apiFetch(method, path, body, headers);
+
+  return new Promise((resolve, reject) => {
+    try {
+      let bytes = 0;
+      res.body.on('data', data => bytes += data.length);
+      res.body.on('error', reject);
+
+      const file = fs.createWriteStream(`${logPath}/${filenamePrefix}.${n.toString().padStart(9, '0')}.dump`);
+      res.body.on('end', () => file.close(() => resolve(bytes)));
+
+      file.on('error', reject);
+
+      res.body.pipe(file);
+    } catch(err) {
+      console.log(err);
+      process.exit(99);
+    }
+  });
+}
+
+async function apiPost(path, body, headers) {
+  const res = await apiFetch('POST', path, body, headers);
+  return res.json();
+}
+
+async function apiFetch(method, path, body, headers) {
+  const url = `${serverUrl}/v1/${path}`;
+
+  const Authorization = bearerToken ? `Bearer ${bearerToken}` : `Basic ${base64(`${userEmail}:${userPassword}`)}`;
+
+  const res = await fetch(url, {
+    method,
+    body,
+    headers: { Authorization, ...headers },
+  });
+  log.debug(method, res.url, '->', res.status);
+  if(!res.ok) throw new Error(`${res.status}: ${await res.text()}`);
+  return res;
+}
+
+function base64(s) {
+  return Buffer.from(s).toString('base64');
+}
+
+function mimetypeFor(f) {
+  const extension = fileExtensionFrom(f);
+  log.debug('fileExtensionFrom()', f, '->', extension);
+  switch(extension) {
+    case 'xls' : return 'application/vnd.ms-excel';
+    case 'xlsx': return 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet';
+    case 'xml' : return 'application/xml';
+    default: throw new Error(`Unsure what mime type to use for: ${f}`);
+  }
+}
+
+function fileExtensionFrom(f) {
+  try {
+    return basename(f).match(/\.([^.]*)$/)[1];
+  } catch(err) {
+    throw new Error(`Could not get file extension from filename '${f}'!`);
+  }
+}
+
+function randomSubmission(n, projectId, formId) {
+  const headers = {
+    'Content-Type': 'multipart/form-data; boundary=foo',
+    'X-OpenRosa-Version': '1.0',
+  };
+
+  const body = `--foo\r
+Content-Disposition: form-data; name="xml_submission_file"; filename="submission.xml"\r
+Content-Type: application/xml\r
+\r
+<data id="250_questions">
```

Comment:
I don't see any advantage of having random values here, only unique instanceID should suffice

Then we can move this to a separate xml file

Reply:
Non-random values will likely affect behaviour and response size for zipped streams.

Reply:
I think this could be moved to a separate file regardless, if that would be helpful.